### DISCUSSION TITLE
Notifications for scheduled maintenance

### DISCUSSION THREAD

**Discussion by maxdanilov** [upptime/upptime#434] [Ideas]
I'd like to propose a feature request for Upptime to be able to automatically announce the upcoming maintenance periods similar to how notifications work for downtimes: https://upptime.js.org/docs/notifications

This is to automate service maintenance windows announcements. It probably makes sense to have a separate provider configuration here (e.g., engineers can get notified in Slack on downtimes, and the customers get upcoming maintenance notifications in Telegram).

Example:
```
<!--
start: 2021-02-24T13:00:00.220Z
end: 2021-02-24T14:00:00.220Z
expectedDown: google, hacker-news
notifyBeforeAt: 2021-02-23T13:00:00.000Z  # (optional) will post an announcement 24 hours before that it's going to happen
notifyAfterAt: 2021-02-24T14:05:00.220Z # (optional) will post an announcement 5 minutes after that it's over
-->
```

**Comment by modem7** [comment]
Having https://github.com/upptime/upptime/discussions/443 tied into this would be useful as well, especially for engineers.
